### DISCUSSION TITLE
SQL certificates

### DISCUSSION THREAD

**Discussion by timofeyp** [strongswan/strongswan#1237] [Q&A] [answered]
How to generate blob format from certificates generated by PKI util, to store in certificates table ?

**Comment by strongX509** [comment, marked as answer]
You can use the `bin2sql` script (https://github.com/strongswan/strongswan/blob/master/scripts/bin2sql.c) to convert X.509 certificates in binary ASN.1-encoded `der` format to the `hex` blob format needed by the SQL certificates table.

Example:

* Certificate `myCert.der` in binary `der` format:
````
od -t x1 myCert.der
0000000 30 82 05 55 30 82 03 3d a0 03 02 01 02 02 08 7c
0000020 24 43 4b b7 dc ef 7e 30 0d 06 09 2a 86 48 86 f7
...
0002500 d6 03 54 9c 77 71 de 4c 15 29 03 2f fb 07 b6 e4
0002520 fe 81 c7 7f a6 3f dc 27 61
````
* Convert the content to SQL `hex` blob:
````
cat myCert.der | ./bin2sql 
X'308205553082033da00302010202087c24434bb7dcef7e300d06092a864886f7...d603549c7771de4c1529032ffb07b6e4fe81c77fa63fdc2761'
````
* You can compile `bin2sql` by changing into the `scripts` directory and executing `make`:
````
cd scripts; make
````
